File: backend/Jarvis.Api/Models/IncomeLog.cs
namespace Jarvis.Api.Models;

public class IncomeLog
{
    public int Id { get; set; }

    public string Source { get; set; } = string.Empty;

    public decimal Amount { get; set; }

    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; }
}

File: frontend/angular-app/src/app/models/income-log.ts
export interface IncomeLog {
  id: number;
  source: string;
  amount: number;
  notes?: string | null;
  createdAt: string;
}

export interface IncomeMonthSummary {
  year: number;
  month: number;
  totalIncome: number;
  entriesCount: number;
  monthlyIncomeGoal: number;
  savingsGoal: number;
  goalProgressPercentage: number;
}
